runtime/lua/vim/lsp/_folding_range.lua

```diff
@@ -5,8 +5,11 @@ local api = vim.api
 
 local M = {}
 
----@class (private) vim.lsp.folding_range.BufState
+---@class (private) FRProvider
```

Comment:
I think it still makes sense to "namespace" the class name with `vim.lsp.folding_range`.